### PR TITLE
chore(travis): add travis to repo

### LINKED ISSUE #3: Add .travis.yml
Linting and running tests on every commit is fun!
Plus all the cool kids are doing it. Wait, you _DO_ want to be cool, don't you?

### PR DESCRIPTION
**Pull request by TDA**
Add travis to repo.
Fixes #3 
@pdehaan
@shane-tomlinson

### CODE REVIEW

**Comment by pdehaan** [comment]
/cc @jrgm

I'm confused as to what our policy is on forcing **npm@2** on all our repos, or if we care.

**Comment by jrgm** [comment]
The npm that comes with node 0.10 (1.4.28) is effectively unsupported and not as reliable as anything npm@2. npm@3 is not out of beta there so don't go there, except of course if you want to try it out in a non-critical way.

**Comment by shane-tomlinson** [comment]
@TDA - can you force the installation of npm@2?

**Comment by shane-tomlinson** [comment]
I'd feel much better about merging #10 if this were merged first.

**Comment by TDA** [comment]
@shane-tomlinson r?

**Comment by pdehaan** [comment]
Anybody have strong opinions on `npm@2` vs `npm@2.4`?
Considering "2.13.1 is the latest of [144 releases](https://www.npmjs.com/package/npm)" I think I'd rather maybe pin to `npm@2` and use the latest "stable" npm 2.x (rather than one that is 6 months old).

> **npm@2.4.1** othiym23 released this on Jan 23 · 734 commits to master since this release
> &mdash; per https://github.com/npm/npm/releases?after=v2.5.0

**Comment by jrgm** [comment]
I don't think anyone suggested 2.4.1 or any specific version. Doing npm@2 is what we should use. (One of these releases will break our travis run someday, but npm has pretty good turnaround on fixing the things (most edge cases) that break in any release, and if we really have to, we can just temporarily pick a non-broken version to get our tests passing again until a fix comes).

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language: node_js
+
+sudo: false
+
+before_install:
+  # Update npm to 2.4
+  - npm install -g npm@2.4
```
Comment (pdehaan) [inline .travis.yml]: @TDA: Can you change this to a generic **npm@2** and we'll grab the latest 2.x.

**Comment by TDA** [comment]
@pdehaan Done, sensei!

**Comment by pdehaan** [comment]
@TDA Can we also add a Travis badge to the README.md file?
Not sure if it will work [yet], or if we'll need @shane-tomlinson's help in enabling Travis for this repo in his Travis dashboard.

**Comment by TDA** [comment]
@pdehaan will do sensei!

**Comment by TDA** [comment]
@pdehaan r?

**Comment by pdehaan** [comment]
I have no powers here. We need an r+ from @shane-tomlinson.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+language: node_js
+
+sudo: false
```
Comment (vladikoff) [inline .travis.yml]: as travis ci blog said we might not need this anymore, it should automatically default to use docker

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+before_install:
+  # Update npm to 2.4
```
Comment (vladikoff) [inline .travis.yml]: remove this comment or say to `2.x`

**Comment by TDA** [comment]
@vladikoff done, and done! :)
